go/adbc/driver/internal/driverbase/rotating_file_writer.go

```diff
@@ -0,0 +1,328 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package driverbase
+
+import (
+	"errors"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+)
+
+const (
+	defaultLogNamePrefix = "apache.adbc.go"
+	defaultFileSizeMaxKb = int64(1024)
+	defaultFileCountMax  = 100
+	defaultTraceFileExt  = ".jsonl"
+)
+
+// Holds the configuration from the options
+type config struct {
+	TracingFolderPath string
+	LogNamePrefix     string
+	FileSizeMaxKb     int64
+	FileCountMax      int
+}
+
+// An option for the RotatingFileWriter
+type rotatingFileWriterOption func(*config)
+
+// Adds the TracingFolderPath option
+func WithTracingFolderPath(tracingFolderPath string) rotatingFileWriterOption {
+	return func(cfg *config) {
+		cfg.TracingFolderPath = tracingFolderPath
+	}
+}
+
+// Adds the LogNamePrefix option
+func WithLogNamePrefix(logNamePrefix string) rotatingFileWriterOption {
+	return func(cfg *config) {
+		cfg.LogNamePrefix = logNamePrefix
+	}
+}
+
+// Adds the FileSizeMaxKb option
+func WithFileSizeMaxKb(fileSizeMaxKb int64) rotatingFileWriterOption {
+	return func(cfg *config) {
+		cfg.FileSizeMaxKb = fileSizeMaxKb
+	}
+}
+
+// Adds the FileCountMax option
+func WithFileCountMax(fileCountMax int) rotatingFileWriterOption {
+	return func(cfg *config) {
+		cfg.FileCountMax = fileCountMax
+	}
+}
+
+func newConfig(options ...rotatingFileWriterOption) (cfg config, err error) {
+	cfg = config{
+		TracingFolderPath: "",
+		LogNamePrefix:     defaultLogNamePrefix,
+		FileSizeMaxKb:     defaultFileSizeMaxKb,
+		FileCountMax:      defaultFileCountMax,
+	}
+	for _, opt := range options {
+		opt(&cfg)
+	}
+	// Ensure default for tracingFolderPath
+	if strings.TrimSpace(cfg.TracingFolderPath) == "" {
+		var fullPath string
+		fullPath, err = getDefaultTracingFolderPath()
+		if err != nil {
+			return
+		}
+		cfg.TracingFolderPath = fullPath
```

Comment:
just do `cfg.TracingFolderPath, err = getDefaultTracingFolderPath()` and get rid of the `fullPath` variable.